### PR TITLE
fix: support server-side pagination for tokens

### PR DESCRIPTION
**Pull request by sh1luo**
Signed-off-by: sh1luo <690898835@qq.com>

### CODE REVIEW

**Comment by sh1luo** [comment]
Demo link: http://1.13.13.178:8000/tokens

File: web/src/TokenListPage.js [hunk 1]
```diff
@@ -217,16 +223,27 @@ class TokenListPage extends React.Component {
       },
     ];
 
+    const paginationProps = {
+      current: this.state.page,
+      pageSize: this.state.pageSize,
+      total: this.state.total,
+      showQuickJumper: true,
+      showSizeChanger: true,
+      showTotal: () => `共${this.state.total}条`,
```
Comment (hsluoyz) [inline web/src/TokenListPage.js:232]: i18n

File: web/src/TokenListPage.js [hunk 2]
```diff
@@ -26,18 +26,24 @@ class TokenListPage extends React.Component {
     this.state = {
       classes: props,
       tokens: null,
+      total: 0,
+      page: 1,
+      pageSize: 5,
```
Comment (hsluoyz) [inline web/src/TokenListPage.js:31]: one state is enough

File: controllers/token.go [hunk 3]
```diff
@@ -28,8 +29,14 @@ import (
...
 	owner := c.Input().Get("owner")
+	limit := util.ParseInt(c.Input().Get("pageSize"))
+	page := util.ParseInt(c.Input().Get("page"))
+	if limit <= 0 || page < 1 {
+		c.ResponseError("Page or limit params error")
+		return
+	}
 
-	c.Data["json"] = object.GetTokens(owner)
+	c.ResponseOk(object.GetTokens(owner, limit, (page-1)*limit))
```
Comment (hsluoyz) [inline controllers/token.go:39]: ?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/token.go (hunk 3 above)
Comment (sh1luo) [reply controllers/token.go:39]: Here need to return array of current page and total number of records. But `c.Data["json"]` only need one interface param.

File: web/src/TokenListPage.js (hunk 2 above)
Comment (sh1luo) [reply web/src/TokenListPage.js:31]: only save total? If that, there will be no loading effect.

**Comment by sh1luo** [comment]
@hsluoyz Done, plz review.

**Pull request closed without merging**